### PR TITLE
migrated to Spring

### PR DESCRIPTION
**Pull request by lilymammadova**

### CODE REVIEW

File: src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("No available apartments found");
+        }
+    }
+
+    @PostMapping("/release/{id}")
+    public ResponseEntity<String> releaseApartment(@PathVariable int id) throws ReservationException {
+        boolean released = apartmentService.release(id);
+        if (released) {
+            return ResponseEntity.ok("Apartment successfully released");
+        } else return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Apartment not found with such id");
```
Comment (Alexmansar) [inline src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java]: } else {
return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Apartment not found with such id");
}

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@lilymammadova Where ourmain method with SpringBootApplication annotation?

**Comment by lilymammadova** [comment]
> @lilymammadova Where ourmain method with SpringBootApplication annotation?

I made first init commit directly to master branch with only main method where SpringBootAnnotation

File: src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java (hunk 1 above)
Comment (lilymammadova) [reply src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java]: fixed

File: src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public ResponseEntity<String> releaseApartment(@PathVariable int id) throws ReservationException {
+        boolean released = hotelService.release(id);
+        if (released) {
+            return ResponseEntity.ok("Apartment successfully released");
+        } else {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Apartment not found with such id");
+        }
+    }
+
+    @GetMapping("/getPaginated")
```
Comment (Alexmansar) [inline src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java]: not good path
just GetMapping could be
or "/filter"

File: src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestControllerAdvice
+public class CustomExceptionHandler {
+
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler(ReservationException.class)
```
Comment (Alexmansar) [inline src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java:14]: 👍

File: src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    public Map<String, String> handleInvalidArgument(ReservationException exception) {
+        Map<String, String> map =  new HashMap<>();
+        map.put("errorMessage", exception.getMessage());
+        return map;
```
Comment (Alexmansar) [inline src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java]: could be just return Map.Of("errorMessage", exception.getMessage())?

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@lilymammadova few small comments

File: src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java (hunk 2 above)
Comment (lilymammadova) [reply src/main/java/org/liliyamammadova/hotelapplication/controller/HotelController.java]: fixed

File: src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java (hunk 4 above)
Comment (lilymammadova) [reply src/main/java/org/liliyamammadova/hotelapplication/exception/CustomExceptionHandler.java]: fixed

**Comment by lilymammadova** [comment]
@Alexmansar I fixed all comments and also added new commit with testcontainers and docker

**Review by Alexmansar — APPROVED** [review]

**Pull request merged**